### PR TITLE
Up and running

### PR DESCRIPTION
**Pull request by hbjorgo**
Basic structure. Added proxy queries (to MainApi) for Areas and Disciplines.

### CODE REVIEW

**Comment by janid1967** [comment]
Starting review

File: src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQuery.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using ServiceResult;
+
+namespace Equinor.Procosys.Library.Query.GetAllDisciplines
+{
+    public class GetAllDisciplinesQuery : IRequest<Result<IEnumerable<DisciplineDto>>>
+    {
+        public GetAllDisciplinesQuery(string plant, IEnumerable<string> classifications)
+        {
+            Plant = plant;
+            Classifications = classifications;
```
Comment (janid1967) [inline src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQuery.cs]: ??new List<string>(); ?

File: src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                .Setup(x => x.CurrentValue)
+                .Returns(options);
+
+            var classifications = new List<string>
+            {
+                "ClassA",
+                "ClassB"
+            };
+
+            _request = new GetAllDisciplinesQuery(Plant, classifications);
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryHandlerTests.cs]: add test with zero classifications. Should get all then

File: src/Equinor.Procosys.Library.Query/GetAllAreas/GetAllAreasQuery.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.Procosys.Library.Query.GetAllAreas
+{
+    public class GetAllAreasQuery : IRequest<Result<IEnumerable<AreaDto>>>
+    {
+        public GetAllAreasQuery(string plant) => Plant = plant;
```
Comment (janid1967) [inline src/Equinor.Procosys.Library.Query/GetAllAreas/GetAllAreasQuery.cs]: require plant

File: src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQuery.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.Procosys.Library.Query.GetAllDisciplines
+{
+    public class GetAllDisciplinesQuery : IRequest<Result<IEnumerable<DisciplineDto>>>
+    {
+        public GetAllDisciplinesQuery(string plant, IEnumerable<string> classifications)
+        {
+            Plant = plant;
```
Comment (janid1967) [inline src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQuery.cs]: require plant

File: src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            _apiVersion = options.CurrentValue.ApiVersion;
+            _baseAddress = new Uri(options.CurrentValue.BaseAddress);
+            _mainApiClient = mainApiClient;
+        }
+
+        public async Task<Result<IEnumerable<DisciplineDto>>> Handle(GetAllDisciplinesQuery request, CancellationToken cancellationToken)
+        {
+            var url = $"{ _baseAddress}Library/Disciplines" +
+                $"?plantId={request.Plant}" +
+                string.Join("", request.Classifications
```
Comment (janid1967) [inline src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs]: Add Classification filter only when any given

File: src/Equinor.Procosys.Library.WebApi/Misc/PlantProvider.cs [hunk 6]
```diff
@@ -6,7 +6,7 @@ namespace Equinor.Procosys.Library.WebApi.Misc
 {
     public class PlantProvider : IPlantProvider
     {
-        public static string PlantHeader = "x-plant";
+        public const string PlantHeader = "x-plant";
 
         private readonly IHttpContextAccessor _accessor;
```
Comment (janid1967) [inline src/Equinor.Procosys.Library.WebApi/Misc/PlantProvider.cs]: format long line 15

File: src/Tests/Equinor.Procosys.Library.Domain.Tests/TimeServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            TimeService.SetProvider(new ValidTimeProvider());
+            var time = TimeService.UtcNow;
+
+            Assert.AreEqual(DateTimeKind.Utc, time.Kind);
+        }
+
+        [TestMethod]
+        public void SetProvider_ThrowsException_WhenProviderIsNull()
+        {
+            Assert.ThrowsException<ArgumentNullException>(() => TimeService.SetProvider(null));
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Domain.Tests/TimeServiceTests.cs]: expr.body. Also line 28

File: src/Tests/Equinor.Procosys.Library.Query.Tests/ClientTests/HttpHelper.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+
+namespace Equinor.Procosys.Library.Query.Tests.ClientTests
+{
+    internal class FakeHttpMessageHandler : DelegatingHandler
+    {
+        private HttpResponseMessage _fakeResponse;
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Query.Tests/ClientTests/HttpHelper.cs]: make readonly

File: src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllAreas/GetAllAreasQueryTests.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿using Equinor.Procosys.Library.Query.GetAllAreas;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Equinor.Procosys.Library.Query.Tests.GetAllAreas
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllAreas/GetAllAreasQueryTests.cs]: add test asserting for ArgumentNullException when plant is null

File: src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryTests.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections.Generic;
+using Equinor.Procosys.Library.Query.GetAllDisciplines;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Equinor.Procosys.Library.Query.Tests.GetAllDisciplines
+{
+    [TestClass]
+    public class GetAllDisciplinesQueryTests
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryTests.cs]: add test asserting for ArgumentNullException when plant is null
Add test asserting for not null but empty list of classifications if not given

File: src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs (hunk 5 above)
Comment (hbjorgo) [reply src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs]: It won't be included if there are no items in the list.

File: src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryHandlerTests.cs [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+            var url = $"{options.BaseAddress}Library/Disciplines" +
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryHandlerTests.cs]: Depending of request contains classifications or not, handler should build 2 different urls.
Add setup and test with zero classifications to ensure urls are built different.  Setup in line 51 and 66 should be duplicated and return different list of MainApiDiscipline depending on built url

File: src/Tests/Equinor.Procosys.Library.WebApi.Tests/Controllers/AreasControllerTests.cs [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    [TestClass]
+    public class AreasControllerTests
+    {
+        private const string Plant = "PCS$TESTPLANT";
+        private Mock<IMediator> _mediatorMock;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            _mediatorMock = new Mock<IMediator>();
```
Comment (janid1967) [inline src/Tests/Equinor.Procosys.Library.WebApi.Tests/Controllers/AreasControllerTests.cs]: expr.body

File: src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs (hunk 5 above)
Comment (janid1967) [reply src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs]: ok, see comment in GetAllDisciplinesQueryHandlerTests.cs regarding testing if no classifications

File: src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllAreas/GetAllAreasQueryTests.cs (hunk 9 above)
Comment (hbjorgo) [reply src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllAreas/GetAllAreasQueryTests.cs]: Do we actually need to fail if a plant name is wrong/empty? Wouldn't it be ok to just return an empty response?
